connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -397,7 +397,7 @@ class SparkConnectPlanner(session: SparkSession) {
   }
 
   private def transformUnresolvedExpression(exp: proto.Expression): UnresolvedAttribute = {
-    UnresolvedAttribute(exp.getUnresolvedAttribute.getUnparsedIdentifier)
+    UnresolvedAttribute.quotedString(exp.getUnresolvedAttribute.getUnparsedIdentifier)
```

Comment:
This is tricky... it's `CatalystSqlParser.parseMultipartIdentifier` v.s. `parseAttributeName` under the hood. Ideally `parseMultipartIdentifier` is more standard but following existing pyspark behavior is more important.

Reply:
> This is tricky... it's `CatalystSqlParser.parseMultipartIdentifier` v.s. `parseAttributeName` under the hood. Ideally `parseMultipartIdentifier` is more standard but following existing pyspark behavior is more important.

This follows not just Pyspark but the Column API in Scala as well. There the string constructor uses exactly this method call.